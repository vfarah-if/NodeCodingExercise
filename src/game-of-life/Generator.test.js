const os = require('os');
const Generator = require('./Generator');

describe('Generator', () => {
	test('should generate an empty two by two board', () => {
		const generator = new Generator(2);

		expect(generator.toString()).toMatchSnapshot();
	});

	test('should generate a two by two board with seeded data in all rows and columns', () => {
		const generator = new Generator(2, [
			{ x: 0, y: 0 },
			{ x: 1, y: 0 },
			{ x: 0, y: 1 },
			{ x: 1, y: 1 },
		]);

		expect(generator.toString()).toMatchSnapshot();
	});

	test('should die from solitide with only one neighbour', () => {
		const generator = new Generator(2, [
			{ x: 0, y: 0 },
			{ x: 1, y: 1 },
		]);

		expect(generator.toString()).toMatchSnapshot('InitialSeed');

		generator.tick();

		expect(generator.toString()).toMatchSnapshot('Tick');
	});

	test('should survive with two or three neighbours', () => {
		const generator = new Generator(3, [
			{ x: 0, y: 0 },
			{ x: 1, y: 1 },
			{ x: 1, y: 2 },
		]);

		expect(generator.toString()).toMatchSnapshot('InitialSeed');

		generator.tick();

		expect(generator.toString()).toMatchSnapshot('Tick');
	});

	test('should generate a four by four block still life', () => {
		const generator = new Generator(4, [
			{ x: 1, y: 1 },
			{ x: 2, y: 1 },
			{ x: 1, y: 2 },
			{ x: 2, y: 2 },
		]);

		expect(generator.toString()).toMatchSnapshot('InitialSeed');

		generator.tick();

		expect(generator.toString()).toMatchSnapshot('Tick');
	});

	test('should generate a three by three blinker oscilator', () => {
		const generator = new Generator(3, [
			{ x: 0, y: 1 },
			{ x: 1, y: 1 },
			{ x: 2, y: 1 },
		]);

		expect(generator.toString()).toMatchSnapshot('InitialSeed');

		generator.tick();

		expect(generator.toString()).toMatchSnapshot('Tick');

		generator.tick();

		expect(generator.toString()).toMatchSnapshot('Tick');

		generator.tick();

		expect(generator.toString()).toMatchSnapshot('Tick');
	});

	test('should generate a six by six blinker oscilator', () => {
		const generator = new Generator(6, [
			{ x: 1, y: 1 },
			{ x: 2, y: 1 },
			{ x: 1, y: 2 },
			{ x: 2, y: 2 },
			{ x: 3, y: 3 },
			{ x: 4, y: 3 },
			{ x: 3, y: 4 },
			{ x: 4, y: 4 },
		]);

		expect(generator.toString()).toMatchSnapshot('InitialSeed');

		generator.tick();

		expect(generator.toString()).toMatchSnapshot('Tick');

		generator.tick();

		expect(generator.toString()).toMatchSnapshot('Tick');

		generator.tick();

		expect(generator.toString()).toMatchSnapshot('Tick');
	});
});
