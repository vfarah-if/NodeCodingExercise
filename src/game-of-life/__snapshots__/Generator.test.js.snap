// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generator should die from solitide with only one neighbour: InitialSeed 1`] = `
" | X |   | 
 |   | X | "
`;

exports[`Generator should die from solitide with only one neighbour: Tick 1`] = `
" |   |   | 
 |   |   | "
`;

exports[`Generator should generate a four by four block still life: InitialSeed 1`] = `
" |   |   |   |   | 
 |   | X | X |   | 
 |   | X | X |   | 
 |   |   |   |   | "
`;

exports[`Generator should generate a four by four block still life: Tick 1`] = `
" |   |   |   |   | 
 |   | X | X |   | 
 |   | X | X |   | 
 |   |   |   |   | "
`;

exports[`Generator should generate a six by six blinker oscilator: InitialSeed 1`] = `
" |   |   |   |   |   |   | 
 |   | X | X |   |   |   | 
 |   | X | X |   |   |   | 
 |   |   |   | X | X |   | 
 |   |   |   | X | X |   | 
 |   |   |   |   |   |   | "
`;

exports[`Generator should generate a six by six blinker oscilator: Tick 1`] = `
" |   |   |   |   |   |   | 
 |   | X | X |   |   |   | 
 |   | X |   |   |   |   | 
 |   |   |   |   | X |   | 
 |   |   |   | X | X |   | 
 |   |   |   |   |   |   | "
`;

exports[`Generator should generate a six by six blinker oscilator: Tick 2`] = `
" |   |   |   |   |   |   | 
 |   | X | X |   |   |   | 
 |   | X | X |   |   |   | 
 |   |   |   | X | X |   | 
 |   |   |   | X | X |   | 
 |   |   |   |   |   |   | "
`;

exports[`Generator should generate a six by six blinker oscilator: Tick 3`] = `
" |   |   |   |   |   |   | 
 |   | X | X |   |   |   | 
 |   | X |   |   |   |   | 
 |   |   |   |   | X |   | 
 |   |   |   | X | X |   | 
 |   |   |   |   |   |   | "
`;

exports[`Generator should generate a three by three blinker oscilator: InitialSeed 1`] = `
" |   |   |   | 
 | X | X | X | 
 |   |   |   | "
`;

exports[`Generator should generate a three by three blinker oscilator: Tick 1`] = `
" |   | X |   | 
 |   | X |   | 
 |   | X |   | "
`;

exports[`Generator should generate a three by three blinker oscilator: Tick 2`] = `
" |   |   |   | 
 | X | X | X | 
 |   |   |   | "
`;

exports[`Generator should generate a three by three blinker oscilator: Tick 3`] = `
" |   | X |   | 
 |   | X |   | 
 |   | X |   | "
`;

exports[`Generator should generate a two by two board with seeded data in all rows and columns 1`] = `
" | X | X | 
 | X | X | "
`;

exports[`Generator should generate an empty two by two board 1`] = `
" |   |   | 
 |   |   | "
`;

exports[`Generator should survive with two or three neighbours: InitialSeed 1`] = `
" | X |   |   | 
 |   | X |   | 
 |   | X |   | "
`;

exports[`Generator should survive with two or three neighbours: Tick 1`] = `
" |   |   |   | 
 | X | X |   | 
 |   |   |   | "
`;
